### PR TITLE
IDEMPIERE-5193 GridTable MaxQueryRecords Limit - Performance

### PR DESCRIPTION
**Pull request by igorpojzl**
Ticket: https://idempiere.atlassian.net/browse/IDEMPIERE-5193

Added Limit on SQL Statement  additional to PreparedStatement setMaxRows() for better performance on Database Servers

### CODE REVIEW

**Comment by igorpojzl** [comment]
Proper testing needed if it does not break something.

File: org.adempiere.base/src/org/compiere/model/GridTable.java [hunk 1]
```diff
@@ -426,6 +426,13 @@ private String createSelectSql()
 		{
 			m_SQL += " ORDER BY " + m_orderClause;
 		}
+		
+		//IDEMPIERE-5193 Add Limit to Query
+		if(m_maxRows > 0)
+		{
+			m_SQL += " FETCH FIRST " + m_maxRows + " ROWS ONLY ";
```
[the remark this answers is not in the document]
Comment (igorpojzl) [reply org.adempiere.base/src/org/compiere/model/GridTable.java]: Fixed, Using DB Pagination.

[the remark this answers is not in the document]
Comment (CarlosRuiz-globalqss) [reply org.adempiere.base/src/org/compiere/model/GridTable.java]: The commit is different to the patch suggested, tested and is not working.

[the remark this answers is not in the document]
Comment (igorpojzl) [reply org.adempiere.base/src/org/compiere/model/GridTable.java]: Sorry, my mistake, fixed

**Comment by igorpojzl** [comment]
@CarlosRuiz-globalqss  Fixed commit.

**Review by hengsin — APPROVED** [review]
looks good.

**Review by CarlosRuiz-globalqss — APPROVED** [review]

**Pull request merged**